### PR TITLE
moved function and struct declarations into a header file

### PR DESCRIPTION
**Pull request by SeppeSoete**
Makes the main.c file a bit less cluttered.

### CODE REVIEW

**Comment by antoine-gannat** [comment]
Nice one ! Way cleaner.

File: src/contagion.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//function declarations
```
Comment (pyinthesky) [inline src/contagion.h:1]: you forgot the includes

```c
#include <stdbool.h> 
#include <stdint.h>
```

Comment (pyinthesky) [reply src/contagion.h:1]: and whichever one defines `gfx_sprite_t`

File: src/contagion.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+typedef struct {
+    //"north america" is 13 characters long, plus 1 for the null terminating byte
+    char name[14];
+    uint8_t *data;
+    uint16_t x;
+    uint8_t y;
+    gfx_sprite_t *spr;
+    //gfx_sprite_t *outlinespr;
+    uint16_t populationdensity;
+    int squares;
```
Comment (pyinthesky) [inline src/contagion.h:66]: 120*160 is 2 bytes, so all these square items can be uint16_t

File: src/contagion.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    int squaresinfected;
+    int squaresdead;
+    int squaresrecovered;
+    //each region has different probability of infection
+    //increase probinfection with mutations
+    uint8_t probinfection;
+} region_t;
+
+typedef struct {
+    uint8_t x;
```
Comment (pyinthesky) [inline src/contagion.h:76]: BUG?: should be uint16_t, as x can be 160?

File: src/contagion.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+//general game data, add more as needed
+typedef struct {
+    long ticksplayed;
+    int numinfected;
+    int numdead;
+    int numrecovered;
+    uint8_t selected;
+    bool canpress;
+    uint8_t toupdate;
+    uint8_t cursorx;
```
Comment (pyinthesky) [inline src/contagion.h:26]: uint16_t

**Pull request merged**